checks/libs/wmi/sampler.py

```diff
@@ -47,18 +47,57 @@ def get(self, key):
         return super(CaseInsensitiveDict, self).get(key.lower())
 
 
+class ProviderArchitectureMeta(type):
+    """
+    Metaclass for ProviderArchitecture.
+    """
+    def __contains__(cls, provider):
+        """
+        Support `Enum` style `contains`.
+        """
+        return provider in cls._AVAILABLE_PROVIDER_ARCHITECTURES
+
+
+class ProviderArchitecture(object):
+    """
+    Enumerate WMI Provider Architectures.
+    """
+    __metaclass__ = ProviderArchitectureMeta
+
+    # Available Provider Architecture(s)
+    DEFAULT = 0
+    _32BIT = 32
+    _64BIT = 64
+    _AVAILABLE_PROVIDER_ARCHITECTURES = frozenset([DEFAULT, _32BIT, _64BIT])
+
+
 class WMISampler(object):
     """
     WMI Sampler.
     """
+    # Properties
+    _provider = None
+    _formatted_filters = None
+
+    # Type resolution state
+    _property_counter_types = None
+
+    # Samples
+    _current_sample = None
+    _previous_sample = None
+
+    # Sampling state
+    _sampling = False
```

Comment:
I agree that it looks better your way :)

The thing we have to be careful about in the future is not to use a mutable value when initializing these properties in the class, i.e. if one day we do something like:

```python
class WMISampler(object):
    _property_counter_types = CaseInsentitiveDict()
    
    def _parse_results(self):
        # ...
        self._property_counter_types[wmi_property.Name] = counter_type
```

then the dict is going to be shared among all the instances.

So I think you can go ahead with your changes but let's be very careful about this behavior in the future :)